### PR TITLE
Fix incomplete array alignment in struct

### LINKED ISSUE #1359: incomplete array in struct alignment is wrong
### Input C/C++ Header

```C++
struct bpf_raw_tracepoint_args {                                                                                              
        unsigned long long args[0];                                                                                                        
};
```

### Bindgen Invocation

```
$ bindgen input.h
```

### Actual Results

```
---- bindgen_test_layout_bpf_raw_tracepoint_args stdout ----
thread 'bindgen_test_layout_bpf_raw_tracepoint_args' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `8`: Alignment of bpf_raw_tracepoint_args', src/main.rs:52:5
```

```rust
#[repr(C)]
#[derive(Default)]
pub struct __IncompleteArrayField<T>(::std::marker::PhantomData<T>);
impl<T> __IncompleteArrayField<T> {
    #[inline]
    pub fn new() -> Self {
        __IncompleteArrayField(::std::marker::PhantomData)
    }
    #[inline]
    pub unsafe fn as_ptr(&self) -> *const T {
        ::std::mem::transmute(self)
    }
    #[inline]
    pub unsafe fn as_mut_ptr(&mut self) -> *mut T {
        ::std::mem::transmute(self)
    }
    #[inline]
    pub unsafe fn as_slice(&self, len: usize) -> &[T] {
        ::std::slice::from_raw_parts(self.as_ptr(), len)
    }
    #[inline]
    pub unsafe fn as_mut_slice(&mut self, len: usize) -> &mut [T] {
        ::std::slice::from_raw_parts_mut(self.as_mut_ptr(), len)
    }
}
impl<T> ::std::fmt::Debug for __IncompleteArrayField<T> {
    fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
        fmt.write_str("__IncompleteArrayField")
    }
}
impl<T> ::std::clone::Clone for __IncompleteArrayField<T> {
    #[inline]
    fn clone(&self) -> Self {
        Self::new()
    }
}
impl<T> ::std::marker::Copy for __IncompleteArrayField<T> {}
#[repr(C)]
#[derive(Debug)]
pub struct bpf_raw_tracepoint_args {
    pub args: __IncompleteArrayField<::std::os::raw::c_ulonglong>,
}
#[test]
fn bindgen_test_layout_bpf_raw_tracepoint_args() {
    assert_eq!(
        ::std::mem::size_of::<bpf_raw_tracepoint_args>(),
        0usize,
        concat!("Size of: ", stringify!(bpf_raw_tracepoint_args))
    );
    assert_eq!(
        ::std::mem::align_of::<bpf_raw_tracepoint_args>(),
        8usize,
        concat!("Alignment of ", stringify!(bpf_raw_tracepoint_args))
    );
    assert_eq!(
        unsafe { &(*(::std::ptr::null::<bpf_raw_tracepoint_args>())).args as *const _ as usize },
        0usize,
        concat!(
            "Offset of field: ",
            stringify!(bpf_raw_tracepoint_args),
            "::",
            stringify!(args)
        )
    );
}
```

### Expected Results
The alignment should be 8, not 1.

### PR DESCRIPTION
**Pull request by photoszzt**
Fix #1359

### CODE REVIEW

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -3658,17 +3658,23 @@ mod utils {
         let incomplete_array_decl = quote! {
             #[repr(C)]
             #[derive(Default)]
-            pub struct __IncompleteArrayField<T>(
-                ::#prefix::marker::PhantomData<T>);
+            pub struct __IncompleteArrayField<T>{
```
Comment (emilio) [inline src/codegen/mod.rs]: nit: there's a space missing here.

File: tests/headers/struct_with_incomplete_array.hpp [hunk 2]
```diff
@@ -0,0 +1,3 @@
+struct bpf_raw_tracepoint_args {                                                                                              
+        unsigned long long args[0];
```
Comment (emilio) [inline tests/headers/struct_with_incomplete_array.hpp]: Could you add a test with another struct with a smaller integer type (like `short`, maybe?).

Comment (photoszzt) [reply tests/headers/struct_with_incomplete_array.hpp]: Hmm, good catch. The alignment is still 8 with the current code when I change to a smaller integer type.

Comment (emilio) [reply tests/headers/struct_with_incomplete_array.hpp]: Oh, I see, for some reason I thought the should always have pointer alignment... I think [T; 0] is the right thing to do, these structs aren't safe to copy generally anyway.

**Comment by photoszzt** [comment]
Hmm, not quite understand the error. Seems that fails at clang not the binding generation.

**Comment by emilio** [comment]
I think with repr(align) support and other fixes that have landed in the meantime like #1592 the issue should be fixed. This also is stale.

Thanks for the patch (and your other fixes) a lot @photoszzt!

**Pull request closed without merging**